packages/babel-helpers/src/helpers.js

```diff
@@ -470,6 +470,16 @@ helpers.wrapNativeSuper = defineHelper(`
   }
 `);
 
+helpers.constructSuperInstance = defineHelper(`
+  export default function _constructSuperInstance(Derived, args, existingThis) {
+    const Super = Derived.__proto__ || Object.getPrototypeOf(Derived);
+    return typeof Reflect !== "undefined" && typeof Reflect.construct === "function" 
+      ? Reflect.construct(Super, args, existingThis.constructor || Derived)
```

Comment:
I also said this in the comment where I suggested this to begin with, but this has the same caveat as the `new.target` transform: the `this.constructor` property _must not be tampered with_, as it's the only hint we have as to what the instance's constructor is in ES5. All other methods available that appear to work at first glance would break if the class is subclassed.